chain/client/src/sync/epoch.rs

```diff
@@ -514,6 +508,148 @@ impl EpochSync {
 
         Ok(())
     }
+
+    // Verify EpochSyncProof
+    fn verify_proof(
```

Comment:
Could we add some tests (can add some to the testloop test) that messing with the proof will result in a failing verification? We could, e.g. check that tweaking every byte of a valid serialized proof will cause a failure one way or another. It doesn't cover all possible attacks but at least makes sure that we aren't forgetting any fields.

Reply:
Created follow-up issue for tests: https://github.com/near/nearcore/issues/12260.
Might need fixing other issues first.